observations/obs_converters/AIRS/convert_amsu_L1.f90

```diff
@@ -107,31 +122,81 @@ program convert_amsu_L1
 filecount = set_filename_list(l1_files, l1_file_list, "convert_amsu_L1")
 
 ! use the first file to initialize the module
-call initialize_amsua_netcdf(l1_files(1))
+call initialize_amsua_netcdf(l1_files(1),verbose)
 
 ! FIXME ... maybe ... are all the files guaranteed to have the
 ! same channels - I think so.
 call channel_list_to_indices(channel_list, use_channels)
 
-! for each input file
+! Figure out how many (more) observations there may be in the output.
+! It is impossible to predict how the lat/lon subsetting may affect this
+! number, but the thinning and channels can be taken into account.
+
+max_num = max_possible_obs(filecount, cross_track_thin, along_track_thin, use_channels)
+
+! either read existing obs_seq or create a new one
+
+inquire(file=outputfile, exist=file_exist)
+
+if ( file_exist .and. append_output ) then
```

Comment:
the original main execution loop in this code was very clean.  could this code (~lines 139-157) be moved to a subroutine?  putting if statements into the main section of code obscures the high level logic.

Reply:
Agreed ... I was going to do that after it passed, and have forgotten. Thanks for the reminder.

Reply:
Done ... the routine is called **append_or_create()** and is in the amsua_netCDF_support_mod.